### ISSUE TITLE
ML: Backtest feedback loop — train → backtest → compare

### ISSUE THREAD

**Issue by edisonshen** [edisonshen/rainier#44]
## Summary

Close the ML feedback loop: train a model, run it through the backtest engine, and compare P&L against the rule-based BookScorer. Without this, all ML training work is blind — CV metrics (accuracy, F1) don't answer "would this model make more money?"

## Problem

Current pipeline ends at offline metrics:
```
features export → train → CV metrics (accuracy, F1, profit_factor) → ???
```

Missing link: **does the trained model actually produce better backtest results than the book scorer?**

### 1. ML-augmented SignalEmitter
- Create a signal emitter that uses MLScorer to set confidence on pin bar signals
- Reuse existing PinBarSignalEmitter pattern detection, replace `score_setup()` with ML probability
- Needs: trained model path, feature extractor, lookback data

### 2. Side-by-side backtest comparison
- Run same historical data through both BookScorer and MLScorer emitters
- Output comparison table:

```
                    BookScorer    MLScorer (v1)    MLScorer (v2)
Total trades           45            38               42
Win rate             53.3%          60.5%            57.1%
Profit factor         1.42           1.78             1.61
Net P&L             $2,340         $3,120           $2,890
Max drawdown        -8.2%          -5.1%            -6.3%
Sharpe ratio          0.85           1.12             0.98
```

### 3. Walk-forward backtest validation
- Train on window 1 → backtest on window 2 → slide forward
- Prevents overfitting: model never sees test data during training
- Report per-window metrics + aggregate

### 4. CLI command
```bash
# Compare book scorer vs ML model on same data
rainier ml compare \
  --csv data/csv/MES_1H.csv \
  --model models/pattern_scorer.json \
  --symbol MES --timeframe 1H

# Walk-forward comparison
rainier ml compare --csv data.csv --model model.json --walk-forward \
  --wf-train-bars 500 --wf-test-bars 100
```

### 5. Per-feature-set comparison
- Compare "all features" vs "selected features" models
- Validates that feature selection actually improved things

## Key design constraints
- Backtest engine is unchanged — only new signal emitters
- Uses existing `run_backtest()` and `BacktestMetrics`
- Walk-forward splits match training splits (no data leakage)
- Comparison report reuses `backtest/report.py` formatting

## Depends on
- #25 Feature store (DONE)
- #26 ScoringStrategy protocol (DONE)
- #28 XGBoost pattern scorer (DONE)
- Feature selector (DONE, PR #43)

## Priority: P1
